presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -285,19 +313,30 @@ private static BytecodeBlock buildInCase(
         BytecodeBlock elseBlock = new BytecodeBlock()
                 .visitLabel(elseLabel);
 
+        BytecodeBlock valueNode = new BytecodeBlock();
+        valueNode.getVariable(value);
+
         Variable wasNull = generatorContext.wasNull();
         if (checkForNulls) {
-            elseBlock.append(wasNull.set(caseWasNull));
+            if (testValues.isEmpty()) {
```

Comment:
Add comment to explain why this is necessary. Add assertion in `generateExpression` that `values.size() > 1` (note: not `arguments`) and explain why the assertion is necessary.